File: demo/hello/java/src/tecgraf/openbus/demo/hello/HelloClient.java
package tecgraf.openbus.demo.hello;

import java.util.Properties;

import openbusidl.rs.IRegistryService;
import openbusidl.rs.Property;
import openbusidl.rs.ServiceOffer;

import org.omg.CORBA.Object;

import scs.core.IComponent;
import tecgraf.openbus.Openbus;
import tecgraf.openbus.exception.OpenBusException;
import demoidl.hello.IHello;
import demoidl.hello.IHelloHelper;

public class HelloClient {
  public static void main(String[] args) throws OpenBusException {
    String userLogin = "tester";
    String userPassword = "tester";

    Properties props = new Properties();
    props.setProperty("org.omg.CORBA.ORBClass", "org.jacorb.orb.ORB");
    props.setProperty("org.omg.CORBA.ORBSingletonClass",
      "org.jacorb.orb.ORBSingleton");
    Openbus bus = Openbus.getInstance();
    bus.resetAndInitialize(args, props, "localhost", 2089);

    IRegistryService registryService = bus.connect(userLogin, userPassword);
    assert (registryService != null);

    Property property = new Property("facets", new String[] { "IHello" });
    ServiceOffer[] servicesOffers =
      registryService.find(new Property[] { property });
    assert (servicesOffers.length == 1);
    ServiceOffer serviceOffer = servicesOffers[0];
    IComponent component = serviceOffer.member;

    Object helloObject = component.getFacetByName("IHello");
    IHello hello = IHelloHelper.narrow(helloObject);
    hello.sayHello();

    bus.disconnect();
    System.out.println("Finalizando...");
  }
}

File: demo/hello/c#/DemoHello_Client/src/HelloClient.cs
using DemoHello_Client.Properties;
using OpenbusAPI;
using OpenbusAPI.Logger;
using openbusidl.rs;
using scs.core;
using System;
using demoidl.hello;

namespace DemoHello_Client
{
  /// <summary>
  /// Cliente do demo hello.
  /// </summary>
  class HelloClient
  {

    static void Main(string[] args) {

      string hostName = DemoConfig.Default.hostName;
      int hostPort = DemoConfig.Default.hostPort;

      Log.setLogsLevel(Level.WARN);

      Openbus openbus = Openbus.GetInstance();
      openbus.Init(hostName, hostPort);

      string userLogin = DemoConfig.Default.userLogin;
      string userPassword = DemoConfig.Default.userPassword;

      IRegistryService registryService = openbus.Connect(userLogin, userPassword);

      string[] facets = new string[] { "IHello" };
      ServiceOffer[] offers = registryService.find(facets);

      if (offers.Length < 1) {
        Console.WriteLine("O serviço Hello não se encontra no barramento.");
        Environment.Exit(1);
      }
      if (offers.Length > 1)
        Console.WriteLine("Existe mais de um serviço Hello no barramento.");

      IComponent component = offers[0].member;
      MarshalByRefObject helloObj = component.getFacetByName("IHello");
      if (helloObj == null) {
        Console.WriteLine("Não foi possível encontrar uma faceta com esse nome.");
        Environment.Exit(1);
      }

      IHello hello = helloObj as IHello;
      if (hello == null) {
        Console.WriteLine("Faceta encontrada não implementa IHello.");
        Environment.Exit(1);
      }

      hello.sayHello();

      openbus.Disconnect();
      openbus.Destroy();

      Console.WriteLine("Fim.");
      Console.ReadLine();
    }
  }

}
